calyx/src/analysis/live_range_analysis.rs

```diff
@@ -234,10 +236,52 @@ impl Debug for LiveRangeAnalysis {
     }
 }
 
+//given a set of shareable and a cell, determines whether cell's
+//type is shareable or not
+fn is_shareable_component(
+    shareable: &HashSet<ir::Id>,
```

Comment:
There are a bunch of functions defined here and inside the `LiveRangeAnalysis` struct that take `shareable: HashSet<ir::Id>` as an parameter. It might make sense to restructure the code so that `shareable` is stored in a struct and have these methods defined on shareable itself. Does that make sense?

Reply:
Like this right? https://doc.rust-lang.org/rust-by-example/fn/methods.html